src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -470,19 +470,13 @@ public bool IsSolutionDPLEnabled
             }
         }
 
-        public bool IsSolutionFullyLoaded
+        public async Task<bool> IsSolutionFullyLoaded()
         {
-            get
-            {
-                return NuGetUIThreadHelper.JoinableTaskFactory.Run(async delegate
-                {
-                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
This may have unexpected side effect when caller switches to main thread from a background thread.